src/avram/delete_operation.cr

```diff
@@ -0,0 +1,234 @@
+require "./validations"
+require "./callbacks"
+require "./define_attribute"
+require "./operation_errors"
+require "./param_key_override"
+require "./inherit_column_attributes"
+require "./needy_initializer_and_delete_methods"
+
+abstract class Avram::DeleteOperation(T)
+  include Avram::NeedyInitializerAndDeleteMethods
+  include Avram::DefineAttribute
+  include Avram::Validations
+  include Avram::OperationErrors
+  include Avram::ParamKeyOverride
+  include Avram::Callbacks
+  include Avram::InheritColumnAttributes
+
+  enum DeleteStatus
+    Deleted
+    DeleteFailed
+    Unperformed
+  end
+
+  macro inherited
+    @@permitted_param_keys = [] of String
+
+    @record : T
+    @params : Avram::Paramable
```

Comment:
The SaveOperation doesn't have these in the `inherited`, but I kept getting an error saying `@record wasn't initialized properly` or whatever. That may be because the SaveOperation has a nilable record, but DeleteOperation shouldn't.